tests/Unit/Parallel/Test_MemoryMonitor.cpp

```diff
@@ -139,8 +173,215 @@ void test_initialize() {
   CHECK(box_tag == empty_holder);
 }
 
+struct TestMetavarsActions {
+  using observed_reduction_data_tags = tmpl::list<>;
+
+  using component_list =
+      tmpl::list<MockMemoryMonitor<TestMetavarsActions>,
+                 MockObserverWriter<TestMetavarsActions>,
+                 SingletonParallelComponent<TestMetavarsActions>,
+                 GroupParallelComponent<TestMetavarsActions>,
+                 ArrayParallelComponent<TestMetavarsActions>,
+                 NodegroupParallelComponent<TestMetavarsActions>>;
+
+  enum class Phase { Initialization, Monitor, Exit };
+
+  void pup(PUP::er& /*p*/) {}
+};
+
+using metavars = TestMetavarsActions;
+using mem_mon_comp = MockMemoryMonitor<metavars>;
+using obs_writer_comp = MockObserverWriter<metavars>;
+using sing_comp = SingletonParallelComponent<metavars>;
+using group_comp = GroupParallelComponent<metavars>;
+using array_comp = ArrayParallelComponent<metavars>;
+using nodegroup_comp = NodegroupParallelComponent<metavars>;
+
+void setup_runner(
+    gsl::not_null<ActionTesting::MockRuntimeSystem<metavars>*> runner) {
+  // Setup all components even if we aren't using all of them
+  ActionTesting::emplace_singleton_component<sing_comp>(
+      runner, ActionTesting::NodeId{1}, ActionTesting::LocalCoreId{1});
+  ActionTesting::emplace_group_component<group_comp>(runner);
+  ActionTesting::emplace_nodegroup_component<nodegroup_comp>(runner);
+  ActionTesting::emplace_array_component<array_comp>(
+      runner, ActionTesting::NodeId{0}, ActionTesting::LocalCoreId{0}, 0);
+
+  // ObserverWriter
+  ActionTesting::emplace_nodegroup_component<obs_writer_comp>(runner);
+  ActionTesting::next_action<obs_writer_comp>(runner, 0);
+  ActionTesting::next_action<obs_writer_comp>(runner, 0);
+
+  // MemoryMonitor
+  ActionTesting::emplace_singleton_component_and_initialize<mem_mon_comp>(
+      runner, ActionTesting::NodeId{2}, ActionTesting::LocalCoreId{2}, {});
+
+  runner->set_phase(metavars::Phase::Monitor);
+}
+
+template <typename Component>
+void check_output(const std::string& filename, const double time,
+                  const size_t num_nodes, const std::vector<double>& sizes) {
+  h5::H5File<h5::AccessType::ReadOnly> read_file{filename + ".h5"};
```

Comment:
we avoid disk IO in these types of tests by having a mock writing action that just records what was meant to be written and then checking that. Disk IO is slow, and just increases likelihood of bugs in general. Eg line 46 of `tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/ResidualMonitorActionsTestHelpers.hpp`